sql/core/src/test/scala/org/apache/spark/sql/DataFrameFunctionsSuite.scala

```diff
@@ -448,6 +448,22 @@ class DataFrameFunctionsSuite extends QueryTest with SharedSQLContext {
       rand(Random.nextLong()), randn(Random.nextLong())
     ).foreach(assertValuesDoNotChangeAfterCoalesceOrUnion(_))
   }
+
+  private def assertNoExceptions(c: Column): Unit = {
+    for (wholeStage <- Seq(true, false)) {
+      withSQLConf((SQLConf.WHOLESTAGE_CODEGEN_ENABLED.key, wholeStage.toString)) {
+        spark.range(0, 5).toDF("a").agg(sum("a")).withColumn("v", c).collect()
```

Comment:
This test also passes without your test. I think you need to reference a `NonDeterministic` expression in the aggregate.

Could also make sure that we test all aggregation paths:

1. HashAggregate
2. ObjectHashAggregate
3. SortAggregate